### PR TITLE
Book Work Hours into Tasks

### LINKED ISSUE #44: Book hours into tasks
Todo: 
- [x] create add button
- [x] create book hour modal
- [x] create/use mutation
- [x] validate that work hour is added in prisma studio

image

### PR DESCRIPTION
**Pull request by YohannesTesfay**
Closes #44 

I added the "comment" but I couldn't be able to see it on Prisma studio.

I think, we would of course still need to implement some validation on the inputs.

### CODE REVIEW

**Comment by YohannesTesfay** [comment]
I tried fixing the 'InputMaybe' from the generated graphql, but no success!

File: app/frontend/components/bookWorkHourModal.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import { format } from 'date-fns'
+import { useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+import { useCreateWorkHourMutation, useProjectsQuery } from '../generated/graphql'
+import { Button } from './button/button'
+import { Modal } from './modal'
+//import { format } from 'date-fns'
```
Comment (PascalHelbig) [inline app/frontend/components/bookWorkHourModal.tsx:7]:
```suggestion
```

File: app/frontend/components/bookWorkHourModal.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      actions={
+        <Button variant="primarySlim" form="book-work-hour" type="submit" disabled={isSubmitting}>
+          Submit
+        </Button>
+      }
+    >
+      <form className="w-full" id="book-work-hour" onSubmit={handleSubmit(handleSubmitHelper)}>
+        <label>
+          Project
+          <select className="w-72 rounded-md" {...register('projectId')}>
```
Comment (PascalHelbig) [inline app/frontend/components/bookWorkHourModal.tsx:73]: Let's add a "required" to the `register` function

File: app/frontend/components/bookWorkHourModal.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+              return (
+                <option value={project.id} key={project.id}>
+                  {project.title}
+                </option>
+              )
+            })}
+          </select>
+        </label>
+        <label>
+          <select className="w-72 rounded-md" {...register('taskId')}>
```
Comment (PascalHelbig) [inline app/frontend/components/bookWorkHourModal.tsx:85]: Let's add a "required" to the `register` function

File: app/frontend/components/bookWorkHourModal.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+              return (
+                <option value={task.id} key={task.id}>
+                  {task.title}
+                </option>
+              )
+            })}
+          </select>
+        </label>
+        <label>
+          <input className="w-72 rounded-md" type="text" {...register('duration', { valueAsNumber: true })} />
```
Comment (PascalHelbig) [inline app/frontend/components/bookWorkHourModal.tsx:96]: Let's add a "required" to the `register` function

Comment (PascalHelbig) [inline app/frontend/components/bookWorkHourModal.tsx:96]: Can we use our own `InputField` component for duration and comment?

**Review by PascalHelbig — CHANGES_REQUESTED** [review]

File: app/pages/time.tsx [hunk 5]
```diff
@@ -90,67 +95,79 @@ const Time = (): JSX.Element => {
...
+        selectedDate={selectedDate}
+        open={isBookWorkHourModalOpen}
+        onClose={() => setIsBookWorkHourModalOpen(false)}
+      />
+      <div className="flex flex-col items-end">
+        <Button variant="primary" onClick={() => setIsBookWorkHourModalOpen(true)}>
+          <BiPlus className="flex items-end text-3xl" />
+        </Button>
+      </div>
+      <ProtectedPage>
```
[the remark this answers is not in the document]
Comment (YohannesTesfay) [reply app/pages/time.tsx:99]: I wonder if it is possible, but I started with `ProtectedPage` and then immediately kept the `Button `component, before the contents of the `ProtectedPage` component.

File: app/frontend/components/bookWorkHourModal.tsx [hunk 6]
```diff
@@ -83,7 +82,7 @@ export const BookWorkHourModal = (props: BookWorkHourModalProps): JSX.Element =>
           </select>
         </label>
         <label>
-          <select className="w-72 rounded-md" {...register('taskId')}>
+          <select className="w-72 rounded-md" {...(register('taskId'), { required: true })}>
```
Comment (YohannesTesfay) [inline app/frontend/components/bookWorkHourModal.tsx:85]: I think adding `required ` is causing problems in loading the tasks in the input. I might of course be missing something, though.

File: app/frontend/components/bookWorkHourModal.tsx (hunk 2 above)
Comment (YohannesTesfay) [reply app/frontend/components/bookWorkHourModal.tsx:73]: Here, if I add "required" to the `register ` function, the projects load, but the tasks don't load any more, unless I am doing it wrong.

**Review by PascalHelbig — APPROVED** [review]

**Comment by YohannesTesfay** [comment]
Closes #44

**Pull request merged**